### ISSUE TITLE
Cannot compile modules unless the --module flag is provided, when the module flag is provided.

### ISSUE THREAD

**Issue by davismj** [Microsoft/TypeScript#7127]
**TypeScript Version:** 

1.8.0-beta (Visual Studio 2015 Community)

**Code**

```ts
// A self-contained demonstration of the problem follows...
export interface Configuration {
  debug?: boolean;
}
```

![image](https://cloud.githubusercontent.com/assets/3845823/13132683/ec30018a-d5c2-11e5-851a-454ebd8b3bc2.png)

**Expected behavior:**
Visual Studio's tsc should be able to tell that the module option is set to system.

**Actual behavior:** 
Visual Studio tells me to do something I've already done.

**Comment by davismj** [comment]
This might just be a misleading error message.

**Comment by billti** [comment]
If you strip down your tsconfig.json to the bare minimum with just the "compilerOptions" and the "module" settings does it still complain? Asking as sometimes an incorrect setting will cause all the tsconfig.json settings to be tossed (and I can't see the whole file in the above). Unfortunately we don't give a very helpful error when this occurs.

**Comment by mhegazy** [comment]
what are the file path to the tsconfig.json, and your config.d.ts files?

**Comment by davismj** [comment]
![image](https://cloud.githubusercontent.com/assets/3845823/13157245/c000df34-d654-11e5-8c5b-a51f907eb259.png)

**Comment by mhegazy** [comment]
can you share your project?

**Comment by mhegazy** [comment]
or a sample project that shows the same symptoms.

**Comment by davismj** [comment]
https://github.com/Foursails/visual-aurelia

Try here.

**Comment by mhegazy** [comment]
what version of the plugin are you using? i am not getting the same issue on 1.8.

**Comment by mhegazy** [comment]
actually i am seeing some errors for missing definitions for Aurelia, but nothing else.

**Comment by davismj** [comment]
1.8.0.0 from here: [https://www.microsoft.com/en-us/download/confirmation.aspx?id=48593](https://www.microsoft.com/en-us/download/confirmation.aspx?id=48593)

Error seems to appear when you try to build the application.

**Comment by billti** [comment]
Are you by any chance defining a `tsconfig.json` but building with the settings in your .csproj file? It is a little confusing, but I see different settings here (and again different for debug and release). See https://github.com/Foursails/visual-aurelia/blob/master/Visual-Aurelia.csproj#L95 .

Perhaps easiest check - if you build just using `tsc.exe` from the command-line in the folder with the `tsconfig.json`, does it work?

**Comment by davismj** [comment]
`tsc.exe --project tsconfig.json` yields

> Files:               1
Lines:           17309
Nodes:           77357
Identifiers:     29805
Symbols:         22229
Types:            4341
I/O read:        0.00s
I/O write:       0.00s
Parse time:      0.14s
Bind time:       0.08s
Check time:      0.31s
Emit time:       0.01s
Total time:      0.54s

And literally no output. What did I do wrong?

**Comment by billti** [comment]
It should find the `tsconfig.json` if you just run `tsc.exe` in the same folder, so no need for the --project switch.

Could you try removing the trailing slashes from rootDir and outDir? I've noticed various issues with some switches if they're not plain folder names (e.g. out of habit I often prefix folder names with "./", and this can mess up some options).

Failing that, I assume you are on Windows from the VS screen shot above? Any weird folder setup that might cause crawling issues (e.g. using redirected folders, network drives, linked files/dirs, etc..?)

**Comment by davismj** [comment]
Removed the trailing slashes, ran tsc.exe without specifying a project, no change.

Nothing funny about this directory structure.

**Comment by mhegazy** [comment]
the tsc.exe --p issue is a duplicate of https://github.com/Microsoft/TypeScript/issues/6841. i have verified that it is fixed in latest.

**Comment by mhegazy** [comment]
the error in VS is a different story.

**Comment by mhegazy** [comment]
i just got it to repro the same error you mentioned above by editing the tsconfig.json file and adding an extra trailing comma. so i believe @billti's analysis is correct.

**Comment by mhegazy** [comment]
i can confirm this is working as expected on latest. TS 1.8.2 should be out in the next few days. closing for now.
